root/etc/e-smith/templates/etc/ssh/sshd_config/70Restricted2Sftp

```diff
@@ -1,23 +1,57 @@
 {
     # restrict groups to sftp, ssh access is refused
+    return "" if ($sssd{'ShellOverrideStatus'} ne 'enabled');
+
     my $domain = $DomainName;
     my @SftpRestricted = ();
+    my @SftpUnRestricted = ();
+
+    my $admins_group = $admins{'group'} || 'domain admins';
 
-    foreach ( split(',',$sshd{'AllowGroups'} || '')) {
+    foreach ( $admins_group, split(',',$sshd{'AllowGroups'} || '')) {
         my ($group, $sftp) = split(':', $_);
-        if(!$group || !$sftp) {
+        if(!$group) {
             next;
         }
+        if (!$sftp) {
+            $sftp = '';
+        }
+
         $group .= "\@$domain" if ($group !~ '@');
-        push @SftpRestricted, $group if ($sftp eq 'sftp');
+        if ($sftp eq 'sftp') {
+            push @SftpRestricted, $group;
+        } else {
+            # do not force SFTP to groups with SSH permission
+            push @SftpUnRestricted, $group;
+       }
     }
 
-    if (@SftpRestricted) {
-        my $matchGroup = join (",",@SftpRestricted);
-        $OUT .= "Match Group \"$matchGroup\"\n";
+   if ($sshd{'AllowEveryone'} eq 'sftp+ssh') {
+        # Everyone is allowed to SSH & SFTP
+        return  "";
+    } 
```

Comment:
This condition can be merged with line 3 and exit early.

```perl
if ( $sssd{'ShellOverrideStatus'} ne 'enabled' || $sshd{'AllowEveryone'} eq 'sftp+ssh') {
    return '';
}
```

Reply:
done